server/services/referal.service.js

```diff
@@ -0,0 +1,72 @@
+import mongoose from 'mongoose';
+import Referral from '../models/referral.model';
+import { ErrorHandler } from '../helpers/errorHandler';
+import httpStatus from '../helpers/httpStatus';
+import { REFERRAL_STATUS } from '../helpers/constants';
+
+const { ObjectId } = mongoose.Types.ObjectId;
+
+export const addReferral = async (referalInfo) => {
+  const info = {
+    userId: referalInfo.userId,
+    refereeId: referalInfo.refereeId,
+    registrationDate: Date.now(),
+  };
+
+  try {
+    const referralInfo = await new Referral(info).save();
+    return referralInfo;
+  } catch (error) {
+    throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error adding referral info', error);
+  }
+};
+
+export const getReferraslByRefereeId = async (refereeId) =>
```

Comment:
Please be mindful of your spellings.
```suggestion
export const getReferralsByRefereeId = async (refereeId) =>
```